controllers/database_reconciler_helpers.go

```diff
@@ -72,71 +72,71 @@ func (r *DatabaseReconciler) handleDelete(ctx context.Context, database *ndbv1al
 	log := ctrllog.FromContext(ctx)
 	log.Info("Database CR is being deleted")
 	if controllerutil.ContainsFinalizer(database, common.FINALIZER_DATABASE_INSTANCE) {
-		// Check if the database instance id (database.Status.Id) is present in the status
-		// If present, then make a deprovisionDatabase API call to NDB
-		// else proceed with removing finalizer as database instance provisioning wasn't successful earlier.
-		if database.Status.Id != "" {
-			infoStatement := "Deprovisioning database instance from NDB."
+		// Check if the deregistration operation id (database.Status.DeregistrationOperationId) is empty
+		// If so, then make a deprovisionDatabase API call to NDB
+		// else proceed check for the operation completion before removing finalizer.
+		deregistrationOperationId := database.Status.DeregistrationOperationId
```

Comment:
Why does an empty DeregistrationOperationId denote deprovisioning the instance from NDB. I know we had this earlier as well, is there a particular reason for this. Is that how the NDB API handles it?

Reply:
We only reach the `handleDelete` function when the user has triggered a deletion of the Database Custom Resource. Now since we know that this resource has to be deleted, we make a deletion/de-registration API call to NDB, and store the deletion operation Id in the database status. 

Since K8s reconcile works in a loop (at least for our operator), this code path will be executed over and over again till the time the resource no longer exists (is deleted). To avoid multiple DELETE /databases/db-id API calls, we check if the deprovisioningOperationId was set in an earlier reconcile (started deletion on NDB already in one of the previous reconcile cycles).

Let me know if this helps.